OpenRA.Game/Server/Server.cs

```diff
@@ -220,8 +220,8 @@ public Server(List<IPEndPoint> endpoints, ServerSettings settings, ModData modDa
 			if (type != ServerType.Local && settings.EnableGeoIP)
 				GeoIP.Initialize();
 
-			if (type != ServerType.Local && UPnP.Status == UPnPStatus.Enabled)
-				UPnP.ForwardPort(Settings.ListenPort, Settings.ListenPort).Wait();
+			if (type != ServerType.Local && Game.Nat != null && Game.Nat.Status == NatStatus.Enabled)
```

Comment:
Recommenting my original feedback that was marked as resolved:

> Relying on all consumers to null-check `Game.UPnP` really isn't great, and is asking for crashes.

The dedicated server NRE proved this point and demonstrates the maintenance burden. Callers shouldn't be forced to jump through hoops to avoid crashes, it would be simpler and safer if they could simply call `Game.TryForwardPort()` which returns false if UPnP is disabled or not working.

Reply:
I tried to avoid `static` for no good reason. `TryForwardPort` seems like a good idea. It also allows for better GUI feedback.